new_app_setup.py

```diff
@@ -0,0 +1,266 @@
+from email.mime.base import MIMEBase
+from email import encoders
+from google.cloud import storage
+
+
+def main():
+
+    base_path = os.path.abspath(os.path.dirname(__file__))
+
+    parser = argparse.ArgumentParser(description='Usage: python3 '
+                                     'new_app_setup.py <Team name> '
+                                     '<List of users>')
+
+    parser.add_argument('team_name', type=str,
+                        help='Team name')
+
+    parser.add_argument('group_user', type=str, help='An email address that '
+                        'will be used for the entire group as a whole.')
+
+    parser.add_argument('-LOU', '--list_of_users',
+                        required=False,
+                        help='A list of the users for this team.')
+
+    parser.add_argument('-RAM', '--RAM', type=int, required=False,
+                        help='How many gigabytes of RAM your team is '
+                        'requesting, the deault is 2G.')
+
+    parser.add_argument('-CPU', '--CPU', type=int, required=False,
+                        help="How much CPU your team is requesting, the "
+                        "default is one.")
+
+    args = parser.parse_args()
+
+    team_name = args.team_name.replace(" ", "-").replace("_", "-").lower()
+
+    namespace_filename = team_name+"_namespace.yml"
+
+    pods_filename = team_name+"_pods.yml"
+
+    create_kubernetes_namespace_yml(namespace_filename, team_name)
+
+    create_kubernetes(namespace_filename, team_name, "namespace")
+
+    if args.RAM is not None:
+        ram_requested = args.RAM
+    else:
+        ram_requested = '2G'  # default
+
+    if args.CPU is not None:
+        cpu_requested = args.CPU
+    else:
+        cpu_requested = '1'  # default
+
+    create_kubernetes_pods_yml(pods_filename, team_name, ram_requested,
+                               cpu_requested)
+
+    create_kubernetes(pods_filename, team_name, "pods")
+
+    config_file = get_congif_yml_file(team_name)
+
+    path_to_congif_file = os.path.join(base_path, config_file)
+
+    send_config_email_to_users(team_name, args.group_user, config_file)
+
+    if args.list_of_users is not None:
+        send_config_email_to_users(team_name, args.list_of_users, config_file)
+
+
+def create_kubernetes_namespace_yml(namespace_filename, team_name):
+
+    tag = "Hack_For_A_Cause_2019"
+
+    yml_file_kubernetes_data = dict(
+
+        kind='Namespace',
+        apiVersion='v1',
+        metadata=dict(
+            name=team_name,
+            labels=dict(
+                name=tag
+                )
+            )
+        )
+
+    with open(namespace_filename, 'w') as outfile:
+        yaml.dump(yml_file_kubernetes_data, outfile, default_flow_style=False)
+
+    print("The YAML file for {} was created and was "
+          "saved as {}".format(team_name, namespace_filename))
+
+    return yml_file_kubernetes_data
+
+
+def create_kubernetes_pods_yml(pods_filename, team_name, ram_requested,
```

Comment:
So then we never set the CPU and RAM for the team? I thought this was a main thing that I was supposed to do for this script, is it not?

Reply:
Yes, one of the main things this should do is set limits. But the limits go on the namespace and are set via a `ResourceQuota`: https://kubernetes.io/docs/tasks/administer-cluster/manage-resources/quota-memory-cpu-namespace/

Once you have a resource quota set on a namespace then all pods must specify how much memory and CPU they want.

For example, let's say the namespace is "example" and we put a `ResourceQuota` on it for 2 CPUs and 2GB of RAM. Then users of that namespace:

1. Create a pod that requests 1 CPU and 1.5 GB of RAM: no problem - k8's will schedule that and they have 1CPU and 1/2 GB of RAM *remaining* in the quota for their namespace
2. Then they try to create a pod that wants 0.5 CPU and 2GB of RAM. k8's will **not** schedule this because the 2GB of RAM is more than what's remaining in their quote
3. The devs then kill the pods in (1). Now there's enough CPU and RAM so the pod from (2) will get run.

In other words: the quota is the total amount of resources that can be used by all the pods in a namespace. Pods request resources which count against that quota.

Reply:
Thank you, your description and link were very helpful!